src/Workspaces/Core/Portable/Workspace/Host/SyntaxTreeFactory/AbstractSyntaxTreeFactoryService.AbstractRecoverableSyntaxRoot.cs

```diff
@@ -91,9 +91,7 @@ public RecoverableSyntaxRoot<TRoot> WithSyntaxTree(IRecoverableSyntaxTree<TRoot>
             {
                 if (this.TryGetValue(out var root))
                 {
-                    var result = new RecoverableSyntaxRoot<TRoot>(_service, root, containingTree);
-                    result._storage = _storage;
-                    return result;
+                    return new RecoverableSyntaxRoot<TRoot>(_service, root, containingTree);
```

Comment:
this explain why it is wrong. https://github.com/dotnet/roslyn/pull/20051#issuecomment-307260122